### PR TITLE
fleet: running fleet via Ignition

### PR DESCRIPTION
**Pull request by dongsupark**
Describe how to set up systemd units for running fleet via rkt, making use of Ignition.

Fixes https://github.com/coreos/bugs/issues/1669

### CODE REVIEW

**Comment by jonboulle** [comment]
Capturing OOB discussion: @dongsupark is going to add the fleet-wrapper script (from https://github.com/coreos/bugs/issues/1669) to the fleet repository, then we will have the ignition config pull down that script so that we don't need to embed a horrible systemd file in-line.

**Comment by jonboulle** [comment]
LGTM but I'd like @crawford to take a pass on it

File: fleet/launching-containers-fleet.md [hunk 1]
```diff
@@ -291,6 +291,40 @@ MachineMetadata=platform=cloud
 MachineMetadata=region=east
 ```
 
+## Running fleet via Ignition
+
+On Container Linux, only fleet 0.11.x is available under /usr/bin. That one might be obsolete for users who want to try out more recent versions. In that case, users should define own custom Ignition configuration to be able to run any version of fleet as they want. For example:
```
Comment (crawford) [inline fleet/launching-containers-fleet.md]: their own

File: fleet/launching-containers-fleet.md [hunk 2]
```diff
@@ -291,6 +291,40 @@ MachineMetadata=platform=cloud
...
+        "source": "https://raw.githubusercontent.com/coreos/fleet/master/scripts/fleet-wrapper"
+      }
+    }]
+  },
+  "systemd": {
+    "units": [
+      {
+        "name": "fleet.service",
+        "enable": true,
+        "contents": "[Unit]\nAfter=etcd.service etcd2.service etcd-member.service\nWants=network.target fleet.socket\n\n[Service]\nType=notify\nRestart=always\nRestartSec=10s\nLimitNOFILE=40000\nTimeoutStartSec=0\nExecStartPre=/usr/bin/mkdir -p /etc/fleet /run/dbus /run/fleet/units\nExecStartPre=/usr/bin/rkt trust --prefix \"quay.io/coreos/fleet\" --skip-fingerprint-review\nExecStart=/opt/bin/fleet-wrapper\n\n[Install]\nWantedBy=multi-user.target"
```
Comment (crawford) [inline fleet/launching-containers-fleet.md]: s/-p/--parents/

**Comment by dongsupark** [comment]
@crawford Fixed. PTAL

**Comment by jonboulle** [comment]
typical

**Comment by jonboulle** [comment]
I tested, and this works, with one caveat: fleet's gonna hang until etcd is started. @dongsupark how about adding a Requires=etcd2.service in this one, since it's a stand-alone example (i.e. without other configuration to make it point to a different etcd)?

**Comment by dongsupark** [comment]
@jonboulle Yeah, right. Pushed one more commit.

**Comment by jonboulle** [comment]
LGTM

**Pull request merged**